Debug test

CezaryGapinski opened the pull request:
This PR is a continuation of @baelter PR #38, after he decided to discontinue usage of this VS Code extension. That really would be a pity to abandon this great feature.

Some minor fixes:
- usage of `--pattern` option for rspec - when pattern option will be added during run, then all test with specified pattern will be run, even if only single test is chosen
- handling `EXT_DIR` environment variable for Windows - the `$EXT_DIR` is not unwrapped correctly, and correct syntax `%EXT_DIR%` must be used
- test line number and test location path in minitests for Windows - in Windows we have path with hard drive letter like `c:\` what cause incorrect received line number handling. Line number is always encoded as a last number in string after colon. Everything before is a file path

thomthom commented on ruby/vscode/minitest.rb:
```diff
@@ -32,6 +32,7 @@ def run(*args)
       reporter = Reporter.new
       reporter.start
       runner = Runner.new(reporter: reporter)
+      puts args
```
Is this intended to be here? Or is this a stray from development/debugging?

CezaryGapinski replied:
@thomthom Thanks for your review!

Actually I also think that puts was only for debugging, that's why removed it. I also revert commit 11f7896 because I didn't find any reason to add everyone ignored files to the official .gitignore in the repo.

Keep in mind that I'm not an author of all commits in this PR, so I will be grateful for any help and review :).

thomthom replied:
I only had a quick glance. I am not familiar with how a VSCode extension work. But I was curious to how it could fix #44. Makes sense to me now that I see it's calling shell commands with environment variable syntax that doesn't work on Windows.

CezaryGapinski commented:
Changes after new push:
- Removed puts args used for debugging purpose in minitest
- Updated changelog
- Added RSpec tests to Rakefile
- Reverted commit 11f7896 - is there any reason to add this additional directory to .gitignore?

pjrashidi commented:
This functionality would be huge for me!

CezaryGapinski commented:
@pjrashidi 
>This functionality would be huge for me! Is there a way I can use this now without waiting for it to get merged? Thank you!

I'm not an expert, but you can try to generate your custom extension.
Install [nodejs](https://nodejs.org/en/) and [git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git).

Run these commands:
`npm install -g typescript`
`npm install -g vsce`
`git clone https://github.com/CezaryGapinski/vscode-ruby-test-adapter.git`
`cd vscode-ruby-test-adapter`
`npm install`
`tsc`
`vsce package`

Go to your vscode, and choose Extensions, click on ellipsis and select "Install from VSIX..." and select generated package.

pjrashidi commented:
Ah, I tried that and it didn't work, I tried it again following your instructions exactly, still didn't work... my test explorer is just blank

Thanks for trying though :D I guess I'll just have to wait for the official release and try again

![image](https://user-images.githubusercontent.com/14811629/85200024-fea03080-b32e-11ea-9e46-ba89eff4d76a.png)

CezaryGapinski commented:
Ok, may I know whether  standard extension from Marketplace works perfectly fine for you and it stop working after you run a custom extension?

Maybe there is a bug in this Pull Request, so I would be appreciate to see your log output with enabled `rubyTestExplorer.logpanel` or `rubyTestExplorer.logfile` option.

pjrashidi commented:
Yeah the standard extension works fine. With this PR version, I don't see anything being logged to output:
![image](https://user-images.githubusercontent.com/14811629/85218832-1207d600-b3d9-11ea-893f-3194e0579227.png)

CezaryGapinski commented:
You should change Output from "Test Explorer" to "Ruby Test Explorer Log" on the right pane :).

Do you use rspec or minitests? Have you tried to change the file pattern to match your test files? I think it is something new in this PR and can cause a problem for rspec, so maybe you have to add pattern like *_spec, spec_* or something else, depend how you named the test files.

pjrashidi commented:
I'm using minitest, but there was no "Ruby Test Explorer" option for the
output pane, I did look for it

On Sun, Jun 21, 2020, 4:39 PM Cezary Gapiński <notifications@github.com>
wrote:

> You should change Output from "Test Explorer" to "Ruby Test Explorer Log"
> on the right pane :).
>
> Do you use rspec or minitests? Have you tried to change the file pattern
> to match your test files? I think it is something new in this PR and can
> cause a problem for rspec, so maybe you have to add pattern like *spec,
> spec* or something else, depend how you named the test files.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/connorshea/vscode-ruby-test-adapter/pull/51#issuecomment-647091946>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADRAD3KDS2DTPLMTI7SUQP3RXW2KDANCNFSM4NOFC2ZQ>
> .
>

CezaryGapinski commented:
>I'm using minitest, but there was no "Ruby Test Explorer" option for the
output pane, I did look for it

I had the same when tsc compiled this extension with errors/warnings (I do not remember exactly how the error looks like), but I had to update the tsc.
You can try to run this extension in debug mode, how it is described in the [contributing section](https://github.com/connorshea/vscode-ruby-test-adapter#contributing).

thomthom commented:
I wonder if there's anything we can do to aid in getting this PR merged?

Would it help if it was broken down into separate PRs per fix? If there's part of this that is easier to accept if they where presented in isolation?
I'm particularly interested in the `EXT_DIR` fix for windows, which looks to be a very small part of whole PR.

connorshea commented:
I'd appreciate if someone could open a new PR (or PRs) with a few changes after I merge this one:

- Add comments for the new functions in this PR that don't have comments, essentially just explaining the purpose of each function.
- Add a setting to optionally disable the debugger functionality entirely, for users who don't want to use it/don't need it.

I won't block this PR on these changes though since none of them are really vital. Thanks so much @CezaryGapinski @baelter, and I'm really sorry it's taken so long to get this functionality merged 🙇

connorshea commented:
Published in 0.8.0 🎉
